src/taipy/core/data/parquet.py

```diff
@@ -0,0 +1,219 @@
+        self,
+        config_id: str,
+        scope: Scope,
+        id: Optional[DataNodeId] = None,
+        name: Optional[str] = None,
+        owner_id: Optional[str] = None,
+        parent_ids: Optional[Set[str]] = None,
+        last_edit_date: Optional[datetime] = None,
+        job_ids: List[JobId] = None,
+        cacheable: bool = False,
+        validity_period: Optional[timedelta] = None,
+        edit_in_progress: bool = False,
+        properties: Dict = None,
+    ):
+        if properties is None:
+            properties = {}
+        if missing := set(self._REQUIRED_PROPERTIES) - set(properties.keys()):
+            raise MissingRequiredProperty(
+                f"The following properties " f"{', '.join(x for x in missing)} were not informed and are required"
+            )
+
+        if self.__ENGINE_PROPERTY not in properties.keys():
+            properties[self.__ENGINE_PROPERTY] = "pyarrow"
+
+        if self.__COMPRESSION_PROPERTY not in properties.keys():
+            properties[self.__COMPRESSION_PROPERTY] = "snappy"
+
+        if self.__READ_KWARGS_PROPERTY not in properties.keys():
+            properties[self.__READ_KWARGS_PROPERTY] = {}
+
+        if self.__WRITE_KWARGS_PROPERTY not in properties.keys():
+            properties[self.__WRITE_KWARGS_PROPERTY] = {}
+
+        self._path = properties.get(self.__PATH_KEY, properties.get(self.__DEFAULT_PATH_KEY))
+        if self._path is None:
+            raise MissingRequiredProperty("default_path is required in a Parquet data node config")
+        else:
+            properties[self.__PATH_KEY] = self._path
+
+        if self.__EXPOSED_TYPE_PROPERTY not in properties.keys():
+            properties[self.__EXPOSED_TYPE_PROPERTY] = self.__EXPOSED_TYPE_PANDAS
+        self._check_exposed_type(properties[self.__EXPOSED_TYPE_PROPERTY])
+
+        super().__init__(
+            config_id,
+            scope,
+            id,
+            name,
+            owner_id,
+            parent_ids,
+            last_edit_date,
+            job_ids,
+            cacheable,
+            validity_period,
+            edit_in_progress,
+            **properties,
+        )
+        if not self._last_edit_date and isfile(self._path):
+            self.unlock_edit()
+
+    @classmethod
+    def storage_type(cls) -> str:
+        return cls.__STORAGE_TYPE
+
+    @property  # type: ignore
+    @_self_reload(DataNode._MANAGER_NAME)
+    def path(self):
+        return self._path
+
+    @path.setter
+    def path(self, value):
+        self._path = value
+        self.properties[self.__PATH_KEY] = value
+
+    def _check_exposed_type(self, exposed_type):
+        if isinstance(exposed_type, str) and exposed_type not in self.__VALID_STRING_EXPOSED_TYPES:
+            raise InvalidExposedType(
+                f"Invalid string exposed type {exposed_type}. Supported values are {', '.join(self.__VALID_STRING_EXPOSED_TYPES)}"
+            )
+
+    def _read(self, read_kwargs: Dict = dict()):
```

Comment:
The read method signature cannot be changed since Taipy scheduler uses it when executing a task. This is the same for `._read()` method though, since it is internal and used by `.read()`.

Therefore, `read_kargs` here is not a good idea.

The `self.properties[self.__EXPOSED_TYPE_PROPERTY]` should be sufficient.

Reply:
Yup I understand that `DataNode.read`'s signature cannot be changed // overriden. 

Having the `read_kwargs` optional param in the `ParquetDataNode._read*` methods is to enable the `.read_with_kwargs` method (the 3rd way of passing arguments to `pandas.read_parquet`) to not have to repeat code by simply passing `read_kwargs` to `._read`. I don't know much about the internal workings, but I currently don't see why `._read` can't have optional params which `.read` need not care about.

If `._read(self)` should be maintained, I'm thinking it might then be better to create `.__read(self, read_kwargs: Dict = dict())` instead. Should I make this change?

Reply:
- The problem here is that the `read` method must stay unchanged since it is in the parent class.
So the parameter passed to the `_read` method is always the default param (empty dict).
That means the `read_kwargs` will not be used when using `read` method. This is a bug.
However, the `read_kwargs` and `write_kwargs` are accessible through the `properties` variable. In particular, you do ensure in the `__init__` method, it is automatically populated even when the user does not specify it. This is good. That means the `_read` method can directly access the `properties` variable. It does not require getting an extra parameter. 

```suggestion
    def _read(self):
        read_kwargs = properties[self.__READ_KWARGS_PROPERTY]
```

At this stage, I don't see the purpose of having another method with the signature:
`.__read(self, read_kwargs: Dict = dict())`
only `_read` will use it.